File: go/stacks/eks-stack.go
package stacks

import (
	"github.com/aws/aws-cdk-go/awscdk/v2"
	"github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	"github.com/aws/aws-cdk-go/awscdk/v2/awseks"
	"github.com/aws/aws-cdk-go/awscdk/v2/awsiam"
	"github.com/aws/constructs-go/constructs/v10"
	// https://docs.aws.amazon.com/sdk-for-go/api/aws/
	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/cdklabs/awscdk-kubectl-go/kubectlv25/v2"

	"k8s-full-stack-cdk-go/helper"
)

type EksNestedStackProps struct {
	awscdk.NestedStackProps
	Vpc awsec2.IVpc
}

func EKSStack(scope constructs.Construct, id string, config helper.Conf, props *EksNestedStackProps) (awscdk.NestedStack, awseks.ICluster) {
	var nsprops awscdk.NestedStackProps
	if props != nil {
		nsprops = props.NestedStackProps
	}
	stack := awscdk.NewNestedStack(scope, &id, &nsprops)

	var cluster awseks.Cluster

	cluster = awseks.NewCluster(stack, aws.String("k8s-sample-cluster"), &awseks.ClusterProps{
		Version: awseks.KubernetesVersion_V1_25(),
		KubectlLayer: kubectlv25.NewKubectlV25Layer(stack, aws.String("kubectl")),
		AlbController: &awseks.AlbControllerOptions{
			Version: awseks.AlbControllerVersion_V2_4_1(),
		},
		DefaultCapacity: aws.Float64(0),
		Vpc: props.Vpc,
		VpcSubnets: &[]*awsec2.SubnetSelection{
			{
				SubnetGroupName: aws.String("eks"),
			},
		},
	})

	cluster.AddNodegroupCapacity(aws.String("eks-nodegroup"), &awseks.NodegroupOptions{
		InstanceTypes: &[]awsec2.InstanceType{
			awsec2.NewInstanceType(aws.String("t3.large")),
		},
	})

	adminUser := awsiam.User_FromUserArn(stack, aws.String("imported_user"), aws.String("arn:aws:iam::<account_id>:user/<iam_user>"))
	cluster.AwsAuth().AddUserMapping(adminUser, &awseks.AwsAuthMapping{
		Groups: &[]*string{
			aws.String("system:masters"),
		},
	})
	

	return stack, cluster
}

File: python/k8s_full_stack_cdk/eks_stack.py
from aws_cdk import (
    # Duration,
    NestedStack,
    aws_eks as eks,
    aws_ec2 as ec2,
    aws_iam as iam
)
from constructs import Construct
from helper import config
from aws_cdk.lambda_layer_kubectl_v25 import KubectlV25Layer

class EKSStack(NestedStack):

    def __init__(self, scope: Construct, construct_id: str, vpc: ec2.IVpc, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        conf = config.Config(self.node.try_get_context('environment'))

        self.cluster = eks.Cluster(
            self, 'k8s-sample-cluster',
            version=eks.KubernetesVersion.V1_25,
            kubectl_layer=KubectlV25Layer(self, "kubectl"),
            alb_controller=eks.AlbControllerOptions(
                version=eks.AlbControllerVersion.V2_4_1
            ),
            default_capacity=0,
            vpc=vpc,
            vpc_subnets=[
                ec2.SubnetSelection(
                    subnet_group_name='eks'
                )
            ]
        )

        self.cluster.add_nodegroup_capacity(
            'eks-nodegroup',
            instance_types=[ec2.InstanceType('t3.large')]
        )

        # importing existing iam user
        admin_user = iam.User.from_user_arn(
            self, 'imported_user',
            user_arn='arn:aws:iam::<account_id>:user/<iam_user>' # change me
        )
        self.cluster.aws_auth.add_user_mapping(admin_user, groups=["system:masters"])



